Packs/FireMonSecurityManager/Integrations/FireMonSecurityManager/FireMonSecurityManager.py

```diff
@@ -0,0 +1,324 @@
+    return result
+
+
+def pca_command(client, args):
+    auth_token = authenticate_command(client)
+    payload = dict(sources=list(args.get("sources").split(",")),
+                   destinations=list(args.get('destinations').split(",")),
+                   services=list(args.get('services').split(",")),
+                   action=args.get('action'), domainId=args.get('domain_id'), deviceGroupId=args.get('device_group_id'))
+    payload_rule_rec = client.rule_rec_api(auth_token, payload)
+    result = {}
+    list_of_device_changes = payload_rule_rec['deviceChanges']
+    for i in range(len(list_of_device_changes)):
+        filtered_rules = []
+        list_of_rule_changes = list_of_device_changes[i]["ruleChanges"]
+        device_id = list_of_device_changes[i]["deviceId"]
+        headers = {'Content-Type': 'application/json',
+                   'accept': 'application/json',
+                   'X-FM-Auth-Token': auth_token}
+
+        for j in range(len(list_of_rule_changes)):
+            if list_of_rule_changes[j]['action'] != 'NONE':
+                filtered_rules.append(list_of_rule_changes[j])
+
+        if filtered_rules is None:
+            return "No Rules Needs to be changed!"
+
+        result[i] = client.validate_pca_change(filtered_rules,
+                                               PCA_URL_SUFFIX.format(args.get('domain_id'), device_id),
+                                               headers)
+        del result[i]['requestId']
+        del result[i]['pcaResult']['startDate']
+        del result[i]['pcaResult']['endDate']
+        del result[i]['pcaResult']['device']['parents']
+        del result[i]['pcaResult']['device']['children']
+        del result[i]['pcaResult']['device']['gpcDirtyDate']
+        del result[i]['pcaResult']['device']['gpcComputeDate']
+        del result[i]['pcaResult']['device']['gpcImplementDate']
+        del result[i]['pcaResult']['device']['state']
+        del result[i]['pcaResult']['device']['managedType']
+        del result[i]['pcaResult']['device']['gpcStatus']
+        del result[i]['pcaResult']['device']['updateMemberRuleDoc']
+        del result[i]['pcaResult']['device']['devicePack']
+        del result[i]['pcaResult']['affectedRules']
+
+    return result
+
+
+def main():
+    verify_certificate = not demisto.params().get('insecure', False)
+    base_url = urljoin(demisto.params()['url'])
+    proxy = demisto.params().get('proxy', False)
+    try:
+        client = Client(
+            base_url=base_url,
+            verify=verify_certificate,
+            proxy=proxy)
+        if demisto.command() == 'test-module':
+            result = test_module(client)
+            demisto.results(result)
+        elif demisto.command() == 'user-authentication':
+            return_results(authenticate_command(client))
+        elif demisto.command() == 'create-pp-ticket':
+            return_results(create_pp_ticket_command(client, demisto.args()))
```

Comment:
When using return_result on main function we need to return from command functions, e.g `create_pp_ticket_command` a CommandResult object. Docs is here
for example:
return CommandResults( outputs_prefix='AnomaliEnterprise.ForensicSearch', outputs_key_field='job_id', outputs=start_search_outputs, readable_output=tableToMarkdown(name="Forensic search started:", t=start_search_outputs, removeNull=True), raw_response=response )